### ISSUE TITLE
searchDelay: blur

### ISSUE THREAD

**Issue by DataTables** [DataTables/DataTables#698]
Suggestion from _jds13_:

How about `searchDelay: blur` to indicate that search should be performed when the search box loses focus?

**Comment by DataTables** [comment]
I quite like this. It might be better to have the return key also trigger a search rather than just a blur event.

**Comment by morrow95** [comment]
+1
May I recommend if this is added to the build that it be extended a bit?

Right now we can call draw with something like : 
```
dt = $('#datatable').DataTable({...});
dt.draw();
```
How about one for search and clear?
```
dt.search();
dt.searchClear();
```

If we were able to customize the search input, those of us using bootstrap in particular, could add two buttons to the search input one for clearing and one for searching.  Would make usability better for the user and I assume would drastically cut down on database hits when using server-side processing. 

If this was possible then maybe something like ` searchDelay: 0 ` could turn off search on key press except for the enter key?

**Comment by BLSully** [comment]
I found this thread while looking for a solution to searchDelay firing on the 'leading edge' of the delay. I am using serverside processing which is expensive to run in this particular case, so I only want to filter after the user has stopped typing for 1000ms. Inspecting the DataTables (1.10.7) source, however, there are no provisions for dropping the leading edge call like underscore provides with their `_.throttle(func, delay, {leading: false})` configuration.

For a quick solution as I really can't afford that initial call when the user starts typing, and in general this would be my preferred default, I've replaced the jqFilter event with the call to the underscore version. Obviously I don't want to be editing the core like this. If the `blur` is added as a feature, could there be an additional option: perhaps `searchDelay` could be turned integrated into `search` like the [other options](http://datatables.net/reference/option/search.smart).
    
    search: {
      'smart': true,
      'delay': 500,
      'type': 'trailing' // 'blur', 'leading' could be other legal values
    }

I have previously always used DataTables in a manner where the search value was only applied locally in the browser, but in my early experiments with serverside processing, I'm finding this is quite the limitation.

**Comment by DataTables** [comment]
Agreed - I'm thinking of changing the behaviour of the delay option entirely in the next major version of DataTables. your idea of using an optional switch is potentially attractive as well - I'll look into that.

**Comment by fabiospampinato** [comment]
I can't really imagine a use case where I type something and expect the search to occur **just** when the input loses focus. IMHO the right answer would simply be to use debouncing, in that case most of the times the search will happen once, a few milliseconds after I finished typing.

Immediately triggering a search when the return key is pressed seems pretty reasonable to me. In the case where debouncing is used with a delay of 350ms and the server returns after another 350ms this will make the process feel twice as fast.

Adding options for further customization of how throttling/debouncing works might be an idea, but not a priority in my opinion, just fixing #810 would make this trivial to implement when using DT in conjunction with lodash/underscore.

Bundling in support for a `clear input` button seems too specific, the API is good enough to implement this without much effort.

If there's still space for improvements to be made before the next major release I'd like if issue #811 and perhaps even #812 could be fixed.
